example/src/App.tsx

```diff
@@ -66,17 +66,20 @@ function Comp2() {
   )
 }
 
-function Comp3() {
-  const t = useControls({ file: { file: undefined } })
+/**
```

Comment:
To bad I can't comment on previous lines, but in Comp1, it seems that nested types in `folder3` (so `t.pos2d`, `t.pos2dArr`, etc) are not properly inferred. Maybe this has to do with the algorithm from use-tweaks?

Reply:
@dbismut Good catch! I've made a [failing](https://github.com/gsimone/leva/pull/36/commits/8f30c3359c879a6e7edfaa1bba69da16f9b466f0) test case for it. Gotta figure out how to make it pass.

Reply:
Ok I'm scratching my head. Somehow nesting a pos2d or 3d array inside a folder causes the type inference to break. I reduced the test case in adfdf69. Any ideas what to do here @dbismut?

Reply:
What's even weirder is that:

```ts
const t1 = useControls({ folder: folder({ pos: [100, 200] }) }) // doesn't work
const f = folder({ pos: [100, 200] })
const t2 = useControls({ folder: f }) // works
```

Reply:
@dbismut I tried flattening the schema in the `folder()` call in 2536293 but the issue persists, though it doesn't break any existing tests. But I think I'm probably mis-using the `Leaves` and `Join` types [here](https://github.com/gsimone/leva/blob/25362935df0603807d8c34965dfafa39592158f0/src/public-api-types.ts#L126).

Reply:
This should be resolved now thanks @AriaMinaei 🎉